### PR TITLE
Support to get transitive groups for authorization

### PR DESCRIPTION
**Pull request by moarychan**
* Support to get transitive groups for authorization, https://github.com/MicrosoftDocs/azure-dev-docs/issues/329.
* Fix spring readme broken link.

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 1]
```diff
@@ -230,6 +240,11 @@ public void validateUserGroupProperties() {
             throw new IllegalArgumentException("One of the User Group Properties must be populated. "
                 + "Please populate azure.activedirectory.user-group.allowed-groups");
         }
+        if (!DEFAULT_GROUP_RELATIONSHIP.equalsIgnoreCase(groupRelationship)
+            && !"transitive".equalsIgnoreCase(groupRelationship)) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:244]: Why don't we have a static variable for this one?

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 2]
```diff
@@ -230,6 +240,11 @@ public void validateUserGroupProperties() {
...
+            throw new IllegalArgumentException("Configuration 'azure.activedirectory.group-relationship' "
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:245]: Is it better we call it xxx.user-group.xxx?

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/ServiceEndpoints.java [hunk 3]
```diff
@@ -44,4 +45,12 @@ public String getAadMembershipRestUri() {
     public void setAadMembershipRestUri(String aadMembershipRestUri) {
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/ServiceEndpoints.java:45]: Is this used anymore?

File: sdk/spring/azure-spring-boot/src/main/resources/service-endpoints.properties [hunk 4]
```diff
@@ -2,15 +2,23 @@ azure.service.endpoints.cn.aadSigninUri=https://login.partner.microsoftonline.cn
 azure.service.endpoints.cn.aadGraphApiUri=https://graph.chinacloudapi.cn/
 azure.service.endpoints.cn.aadKeyDiscoveryUri=https://login.partner.microsoftonline.cn/common/discovery/keys
 azure.service.endpoints.cn.aadMembershipRestUri=https://graph.chinacloudapi.cn/me/memberOf?api-version=1.6
+azure.service.endpoints.cn.aadTransitiveMemberRestUri=https://graph.chinacloudapi.cn\
+  /me/transitiveMemberOf?api-version=1.6
 azure.service.endpoints.cn-v2-graph.aadSigninUri=https://login.partner.microsoftonline.cn/
 azure.service.endpoints.cn-v2-graph.aadGraphApiUri=https://microsoftgraph.chinacloudapi.cn/
 azure.service.endpoints.cn-v2-graph.aadKeyDiscoveryUri=https://login.partner.microsoftonline.cn/common/discovery/keys
 azure.service.endpoints.cn-v2-graph.aadMembershipRestUri=https://microsoftgraph.chinacloudapi.cn/v1.0/me/memberOf
+azure.service.endpoints.cn-v2-graph.aadTransitiveMemberRestUri=https://microsoftgraph.chinacloudapi.cn\
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/resources/service-endpoints.properties:11]: why break the line here

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:244]: Will add another one.

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java (hunk 2 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:245]: Okay

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/ServiceEndpoints.java (hunk 3 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/ServiceEndpoints.java:45]: no, will use it by default

File: sdk/spring/azure-spring-boot/src/main/resources/service-endpoints.properties (hunk 4 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot/src/main/resources/service-endpoints.properties:11]: because the checkstyle is not passed

**Review by backwind1233 — APPROVED** [review]
LGTM

**Pull request merged**